### ISSUE TITLE
vkGetDeviceQueue is not strong enough

### ISSUE THREAD

**Issue by eranzhao** [KhronosGroup/Vulkan-LoaderAndValidationLayers#146]
The vkGetDeviceQueue is not strong enough, when the _queueFamilyIndex_ or _queueIndex_ is out of range, there's only null accessing message generated but no more information:

```
0xC0000005: Access violation reading location 0x00000000 (call stack: _vkGetDeviceQueue@16())
```

I think the vkGetDeviceQueue should not modify the value of _pQueue_ when the index is out of range, instead of null accessing.

**Comment by courtney-g** [comment]
Are you seeing this error while running the Validation layers?

**Comment by lentinem** [comment]
I disagree. If you are running validation then the behavior should not change the behavior unless you tell it to do so. I would also have expected param checker to throw an error in this case so I'm guessing you are not in which case I also don't think the driver should be responsible for this as one of the major points of the layers is to catch app issues so the driver doesn't have to.

**Comment by dustin-lunarg** [comment]
Enabling standard validation should produce the following errors when the queueIndex is out of range:

>ERROR: [PARAMCHECK] Code 1 : VkGetDeviceQueue parameter, uint32_t queueIndex 99, must be less than the number of queues given when the device was created.
>ERROR: [DL] Code 9 : Invalid queue request in vkGetDeviceQueue(). QueueFamilyIndex 0 only has 1 queues, but requested queueIndex is 99.

You can find instructions for enabling validation layers in the layers [README](https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/blob/master/layers/README.md) file.

**Comment by lentinem** [comment]
Closing due to inactivity
